### PR TITLE
release: add prompt to draft GitHub release

### PR DESCRIPTION
**Pull request by ivanvc**
This is a cherry-pick from #18886. This pull request can be merged soon without adding the automated step to undraft/release the GitHub release.

We identified the potential improvement while doing the 3.5.17 / 3.4.35 releases.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -337,6 +337,9 @@ main() {
     log_warning "WARNING: If not running on DRY_MODE, please do the GitHub release manually."
     log_warning ""
   else
+    read -p "Create GitHub draft release in ${REPOSITORY} [y/N]? " -r confirm
+    [[ "${confirm,,}" == "y" ]] || exit 1
```
[the remark this answers is not in the document]
Comment (ivanvc) [reply scripts/release.sh:341]: I copied the code/model from previous prompts. It's also the last step, so a failure exit here prevents nothing from later steps.

I'm okay changing it not to exit. Let me know what you think.

[the remark this answers is not in the document]
Comment (ahrtr) [reply scripts/release.sh:341]: A user's valid input should NOT lead to release failure.

Then why add this prompt? In what cases we may select "Y" or "N"?

I think we should
- either do not add this prompt question at all; which means we always create draft github release;  [this option is better?]
- or continue to execute if users input "N"; and provide a warning

[the remark this answers is not in the document]
Comment (ivanvc) [reply scripts/release.sh:341]: That's a good point. I added this to be consistent with other steps in the release script. I'm fine dropping this pull request. Or with your latest suggested option. @jmhbnz, WDYT?

[the remark this answers is not in the document]
Comment (ivanvc) [reply scripts/release.sh:341]: I'm dropping this pull request and keeping the current behavior. Please chime in if you think otherwise.

**Comment by ivanvc** [comment]
Closed to keep the current behavior. Refer to https://github.com/etcd-io/etcd/pull/18895#discussion_r1842777313.

**Pull request closed without merging**